src/strategies/pod-leader/index.ts

```diff
@@ -0,0 +1,518 @@
+
+  printLog();
+
+  const tokenCount = poolStaked
+    .mul(tokensPerLp)
+    .div(tokenDecimals)
+    .mul(weight)
+    .div(weightDecimals);
+
+  log.push(`tokenCount = ${tokenCount}`);
+
+  return toFloat(tokenCount, _options.decimals) * price;
+}
+
+function printLog() {
+  if (_options.log || false) {
+    console.debug(log);
+    log = [];
+  }
+}
+
+async function getTokenPrice(
+  tokenAddress: any,
+  network: any,
+  provider: any,
+  blockTag: string | number
+) {
+  let price = 1;
+  const cacheKey = tokenAddress + blockTag;
+
+  if (_options.usePrice === true && !priceCache.has(cacheKey)) {
+    log.push(
+      `calling getPrice for token address: ${tokenAddress} and blockTag: ${blockTag}`
+    );
+
+    priceCache.set(
+      cacheKey,
+      (await getPrice(network, provider, tokenAddress, blockTag)) || 1
+    );
+  }
+
+  price = priceCache.get(cacheKey) || 1;
+
+  return price;
+}
+
+async function getPrice(network, provider, address, blockTag) {
+  if (!blockCache.has(blockTag)) {
+    blockCache.set(blockTag, await provider.getBlock(blockTag));
+  }
+
+  const block = blockCache.get(blockTag);
+  const platform = networksWithPlatforms[network];
+  const currency = _options.currency || 'usd';
+  const from = block.timestamp - 100000;
+  const to = block.timestamp;
+  const coingeckoApiURL = `https://api.coingecko.com/api/v3/coins/${platform}/contract/${new String(
+    address
+  ).toLowerCase()}/market_chart/range?vs_currency=${currency}&from=${from}&to=${to}`;
+
+  log.push(`platform = ${platform}`);
+  log.push(`from = ${from}`);
+  log.push(`to = ${from}`);
+  log.push(`coingeckoApiURL = ${coingeckoApiURL}`);
+
+  const coingeckoData = await fetch(coingeckoApiURL)
+    .then(async (r) => {
+      log.push(`coingeco response = ${JSON.stringify(r, undefined, 2)}`);
+
+      const json = await r.json();
+
+      log.push(`coingecko json = ${JSON.stringify(json, undefined, 2)}`);
+
+      return json;
+    })
+    .catch((e) => {
+      console.error(e);
+      throw new Error(
+        'Strategy masterchef-pool-balance-of-token: coingecko api failed'
+      );
+    });
+
+  return (coingeckoData.prices?.pop()?.pop() || 0) as number;
+}
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  _options = options;
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const userInfoCalls = getUserInfoCalls(addresses);
+  const tokenCalls = getTokenCalls();
+  const entries = new Map<PropertyKey, any>();
+
+  const userInfoResponse = await multicall(
+    network,
+    provider,
+    abi,
+    userInfoCalls,
+    { blockTag }
+  );
+
+  const userInfoChunks = arrayChunk(userInfoResponse, 1);
+
+  const tokenResponse = await multicall(network, provider, abi, tokenCalls, {
+    blockTag
+  });
+
+  for (let i = 0; i < userInfoChunks.length; i++) {
+    const value = userInfoChunks[i];
+    const score = await processValues(
```

Comment:
Here we are calling `processValues` function for every address, which has multiple fetch calls (ex: getPrice), and it is not scalable in the case of a large number of voters, is there an alternative you can think of where you can get data from a single call?

 Here is the checklist to add any strategy:
https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy

Reply:
I have removed the getPrice to simplify it, just based off token weighting now. Should be able to scale properly now.